File: CA.ERP.WebApp/ClientApp/src/app/home/purchase-order/supplier/models/supplier-view.ts
import { SupplierBrand } from './supplier-brand';

export interface SupplierView {
  id: string;
  status: number;
  createdAt: string;
  createdBy: string;
  updatedAt: string;
  updatedBy: string;
  name: string;
  address: string;
  contactPerson: string;
  supplierBrands: SupplierBrand[];
}

File: CA.ERP.WebApp/Dto/Supplier/SupplierView.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CA.ERP.WebApp.Dto.Supplier
{
    /// <summary>
    /// Supplier Dto
    /// </summary>
    public class SupplierView : DtoViewBase
    {
        /// <summary>
        /// The name of the supplier.
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Adress of the supplier
        /// </summary>
        public string Address { get; set; }
        /// <summary>
        /// The name of the contact person for the supplier
        /// </summary>
        public string ContactPerson { get; set; }

        /// <summary>
        /// Brands available to the supplier
        /// </summary>
        public List<SupplierBrandView> SupplierBrands { get; set; } = new List<SupplierBrandView>();
    }
}